### PR TITLE
Adding TesterMain.f90

### PR DESCRIPTION
**Pull request by pclausen**
TesterMain.f90 is a custom-module built for testing of execism.io fortran track exercises built with CMake.

@kytrinyx : I have added the MIT license to this fortran file. It might be unnecessary as LICENSE file in root clearly identifies this, but I was uncertain so I added it - it is easy to delete; let me know.

### CODE REVIEW

File: testlib/TesterMain.f90 [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+! AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+! LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+! OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+! SOFTWARE.
+!------------------------------------------------------------------
+!------------------------------------------------------------------
+!
+! USAGE:
+!
+! This is a custom-module built for testing of execism.io fortran track exercises.
```
Comment (SaschaMann) [inline testlib/TesterMain.f90]:
```suggestion
! This is a custom-module built for testing of exercism.io Fortran track exercises.
```

File: testlib/TesterMain.f90 [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+!
+! It should be used in test-code with 
+!
+!   call assert_equal( input, expected, message)
```
Comment (SaschaMann) [inline testlib/TesterMain.f90]:
```suggestion
!   call assert_equal(input, expected, message)
```

File: testlib/TesterMain.f90 [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+  end function
+  
+! Double precision to string
+  function d_to_s(d)
+    double precision, intent(in) :: d
+    character(len=MAX_STRING_LEN) :: d_to_s
+    write(d_to_s, *) d
+  end function
+
+! Logical/bolean to string
```
Comment (SaschaMann) [inline testlib/TesterMain.f90]:
```suggestion
! Logical/boolean to string
```

File: testlib/TesterMain.f90 [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+  function b_to_s(b)
+    logical, intent(in) :: b
+    character(len=5) :: b_to_s
+    if (b) then
+      b_to_s='True '
```
Comment (SaschaMann) [inline testlib/TesterMain.f90]:
```suggestion
      b_to_s='True'
```

Comment (pclausen) [reply testlib/TesterMain.f90]: the space is intensional as the string is fixed length = 5

Comment (SaschaMann) [reply testlib/TesterMain.f90]: Ah, alright

**Pull request merged**